src/main/java/seedu/address/model/tag/DifficultyTag.java

```diff
@@ -37,6 +37,8 @@ public class DifficultyTag extends Tag {
      * @return true if the difficulty level is "low", "medium", or "high".
      */
     private static boolean isValidDifficulty(String difficultyLevel) {
+        assert difficultyLevel != null : "difficultyLevel should not be null";
```

Comment:
Great job on using assertions!

Reply:
Could consider using the assert not null method from JUnit